### PR TITLE
Remove validation for embargo date if embargo option is not selected

### PR DESCRIPTION
**Pull request by jefferya**
Fixes: #996 

Purpose: improve validation to handle case where `visibility: embargo` is not set and the embargo date is present. 

Changes:
- Remove validation for embargo date if embargo option is not set
- Add test for failure when object has visibility open combined with an embargo date

### CODE REVIEW

File: app/models/concerns/item_properties.rb [hunk 1]
```diff
@@ -100,7 +100,6 @@ def as_json(_options)
       # If you're looking for rights and subject validations, note that they have separate implementations
       # on the Thesis and Item classes.
       validates :embargo_end_date, presence: true, if: ->(item) { item.visibility == VISIBILITY_EMBARGO }
-      validates :embargo_end_date, absence: true, if: ->(item) { item.visibility != VISIBILITY_EMBARGO }
       validates :visibility_after_embargo, presence: true, if: ->(item) { item.visibility == VISIBILITY_EMBARGO }
```
Comment (jefferya) [inline app/models/concerns/item_properties.rb]: The embargo date validation differs from draft item to item models. I'm not sure why?

File: test/models/draft_item_test.rb [hunk 2]
```diff
@@ -167,6 +167,12 @@ def before_all
     )
 
     assert draft_item.valid?
+
+    draft_item.assign_attributes(
+      visibility: DraftItem.visibilities[:open_access]
+    )
+
+    assert draft_item.valid?
```
Comment (jefferya) [inline test/models/draft_item_test.rb]: ToDo: This test should also be updated in test/models/item_test.rb

**Comment by weiweishi** [comment]
@jefferya see you've removed WIP. Is this ready to be reviewed?

File: test/models/draft_item_test.rb [hunk 3]
```diff
@@ -167,6 +167,12 @@ def before_all
     )
 
     assert draft_item.valid?
+
+    draft_item.assign_attributes(
```
Comment (ConnorSheremeta) [inline test/models/draft_item_test.rb]: There should probably be a similar check like this in /test/models/draft_thesis_test.rb

**Comment by ConnorSheremeta** [comment]
No commits have been made since my comments. The error is still present for theses and I believe there should be a test added in /test/models/draft_thesis_test.rb

**Pull request merged**